sgkit/distance/metrics.py

```diff
@@ -172,3 +175,234 @@ def correlation_reduce(v: ArrayLike, out: ArrayLike) -> None:  # pragma: no cove
     if denom > 0:
         value = 1 - (num / denom)
     out[0] = value
+
+
+def call_metric_kernel(
+    f: ArrayLike, g: ArrayLike, metric: str, metric_kernel: Any
+) -> ArrayLike:  # pragma: no cover.
+    # Numba's 0.54.0 version is required, which is not released yet
+    # We install numba from numba conda channel: conda install -c numba/label/dev numba
+    # Relevant issue https://github.com/numba/numba/issues/6824
+    f = np.ascontiguousarray(f)
+    g = np.ascontiguousarray(g)
+
+    # move input data to the device
+    d_a = cuda.to_device(f)
+    d_b = cuda.to_device(g)
+    # create output data on the device
+    out = np.zeros((f.shape[0], g.shape[0], N_MAP_PARAM[metric]), dtype=f.dtype)
+    d_out = cuda.to_device(out)
+
+    # TODO: Consider using cupy for directly creating zeros on GPU
+    # d_out = cp.zeros((f.shape[0], g.shape[0], N_MAP_PARAM[metric]), dtype=f.dtype)
+
+    threads_per_block = (32, 32)
```

Comment:
How good do you feel about this as a default?  Did this come from somewhere else and do you think is it likely to matter much on different hardware?

Reply:
Quite good I would say, this is the maximum number of threads you can have in a block 

https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#features-and-technical-specifications__technical-specifications-per-compute-capability

These apply to compute capability 2.0 and higher and all GPUs NVIDIA has shipped in the past 10+ years have compute capability > 3.0 as far as I know.

One way to get compute capability programmatically is: 

```python
In [1]: from numba import cuda
In [2]: device = cuda.get_current_device()
In [3]: device.compute_capability
Out[3]: (7, 5)
```

The above is for the Quansight GPU, I was working on, do you think I should add this check in the code rather?

Reply:
Nope, no need for the check as far as I'm concerned.  Thanks for those details!

All set on this point.